src/glmfit.jl

```diff
@@ -758,3 +874,50 @@ function checky(y, d::Binomial)
     end
     return nothing
 end
+
+function nobs(r::GlmResp{V,D,L,W}) where {V,D,L,W<:AbstractWeights}
+    oftype(sum(one(eltype(r.wts))), length(r.y))
+end
+nobs(r::GlmResp{V,D,L,W}) where {V,D,L,W<:FrequencyWeights} = sum(r.wts)
+
+function residuals(r::GlmResp; weighted::Bool=false)
+    y, η, μ = r.y, r.eta, r.mu
+    dres = similar(μ)
+
+    @inbounds for i in eachindex(y, μ)
+        μi = μ[i]
+        yi = y[i]
+        dres[i] = sqrt(max(0, devresid(r.d, yi, μi))) * sign(yi - μi)
+    end
+
+    if weighted
+        dres .*= sqrt.(r.wts)
+    end
+
+    return dres
+end
+
+function momentmatrix(m::GeneralizedLinearModel)
+    X = modelmatrix(m; weighted=false)
+    r = varstruct(m)
+    m.rr.d isa Union{Gamma, InverseGaussian} && (r .= r.*(sum(working_weights(m)) / sum(abs2, r)))
+    return Diagonal(r) * X
+end
+
+function varstruct(x::GeneralizedLinearModel)
+    wrkwt = working_weights(x)
+    wts = weights(x)
+    wrkwts = wts isa ProbabilityWeights ? wrkwt .* (nobs(x) ./ wts.sum) : wrkwt
```

Comment:
```suggestion
    wrkwts = wts isa ProbabilityWeights ? wrkwt .* (nobs(x) / wts.sum) : wrkwt
```

Reply:
Also, using both `wrkwt` and `wrkwts` in the same function doesn't sound like a good idea. Using a single name is fine here.